src/components/UniprotInput.vue

```diff
@@ -45,7 +47,22 @@ export default Vue.extend({
   computed: {},
   watch: {
     searchQuery() {
-      this.triggerQuery();
+      if (!this.skipQueryTrigger) {
```

Comment:
How about `if (this.searchQuery !== this.protein.uniprotId) {`

Reply:
Will work with a little change:
```
searchQuery() {
      if (
        this.passedProtein &&
        this.searchQuery === this.passedProtein.uniprotId
      ) {
        return;
      }
      this.triggerQuery();
    },
```